dash/dash.py

```diff
@@ -234,6 +234,12 @@ def layout(self, value):
 
         self._layout = value
 
+        self._components = {}
```

Comment:
We keep a dictionary of all the component instances which can have their props changed (e.g. components with the `id` field. Dash did not have a way to reference the class or the instance of the component associated with an (`id`, `prop`) combo before.